client/asset/btc/btc.go

```diff
@@ -4739,6 +4925,474 @@ func (btc *intermediaryWallet) EstimateSendTxFee(address string, sendAmount, fee
+		newReserves, unspent = btc.bondSpent(amt)
+		btc.log.Debugf("New bond reserves (abandoned post) = %f BTC with %f in unspent bonds",
+			toBTC(newReserves), toBTC(unspent)) // increment/restore and report new
+		err := btc.ReturnCoins(coins)
+		if err != nil {
+			btc.log.Errorf("error returning coins for unused bond tx: %v", coins)
+		}
+	}
+
+	var success bool
+	defer func() {
+		if !success {
+			abandon()
+		}
+	}()
+
+	totalIn, _, err := btc.addInputsToTx(baseTx, coins)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to add inputs to bond tx: %w", err)
+	}
+
+	changeAddr, err := btc.node.changeAddress()
+	if err != nil {
+		return nil, nil, fmt.Errorf("error creating change address: %w", err)
+	}
+	signedTx, _, _, err := btc.signTxAndAddChange(baseTx, changeAddr, totalIn, amt, feeRate)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to sign bond tx: %w", err)
+	}
+
+	txid := signedTx.TxHash()
+
+	signedTxBytes, err := serializeMsgTx(signedTx)
+	if err != nil {
+		return nil, nil, err
+	}
+	unsignedTxBytes, err := serializeMsgTx(baseTx)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Prep the redeem / refund tx.
+	redeemMsgTx, err := btc.makeBondRefundTxV0(&txid, 0, amt, bondScript, bondKey, feeRate)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to create bond redemption tx: %w", err)
+	}
+	redeemTx, err := serializeMsgTx(redeemMsgTx)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to serialize bond redemption tx: %w", err)
+	}
+
+	bond := &asset.Bond{
+		Version:    ver,
+		AssetID:    btc.cloneParams.AssetID,
+		Amount:     amt,
+		CoinID:     toCoinID(&txid, 0),
+		Data:       bondScript,
+		SignedTx:   signedTxBytes,
+		UnsignedTx: unsignedTxBytes,
+		RedeemTx:   redeemTx,
+	}
+	success = true
+
+	return bond, abandon, nil
+}
+
+func (btc *baseWallet) makeBondRefundTxV0(txid *chainhash.Hash, vout uint32, amt uint64,
+	script []byte, priv *secp256k1.PrivateKey, feeRate uint64) (*wire.MsgTx, error) {
+	lockTime, pkhPush, err := dexbtc.ExtractBondDetailsV0(0, script)
+	if err != nil {
+		return nil, err
+	}
+
+	pk := priv.PubKey().SerializeCompressed()
+	pkh := btcutil.Hash160(pk)
+	if !bytes.Equal(pkh, pkhPush) {
+		return nil, fmt.Errorf("incorrect private key to spend the bond output")
+	}
+
+	msgTx := wire.NewMsgTx(btc.txVersion())
+	// Transaction LockTime must be <= spend time, and >= the CLTV lockTime, so
+	// we use exactly the CLTV's value. This limits the CLTV value to 32-bits.
+	msgTx.LockTime = lockTime
+	bondPrevOut := wire.NewOutPoint(txid, vout)
+	txIn := wire.NewTxIn(bondPrevOut, []byte{}, nil)
+	txIn.Sequence = wire.MaxTxInSequenceNum - 1 // not finalized, do not disable cltv
+	msgTx.AddTxIn(txIn)
+
+	// Calculate fees and add the refund output.
+	var outputSize int
+	if btc.segwit {
+		outputSize = dexbtc.P2WPKHOutputSize
+	} else {
+		outputSize = dexbtc.P2PKHOutputSize
+	}
+	redeemSize := msgTx.SerializeSize() + dexbtc.RedeemBondSigScriptSize + outputSize
```

Comment:
`dexbtc.RedeemBondSigScriptSize` does not apply to segwit, where the sigscript would be 1+scripthash, and the witness data would contain the signatures and script itself.  We'll have to do all the **vbytes** computations to be precise about this. `MsgTxVBytes` or like https://github.com/decred/dcrdex/blob/4d5c1cc7fa25b46bcb3c4565f69d25f93c0f8071/client/asset/btc/btc.go#L3550-L3558?